cmake/modules/platform/toolcfg/xlc.cmake

```diff
@@ -19,30 +19,145 @@
 # SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
 ###############################################################################
 
-set(OMR_WARNING_AS_ERROR_FLAG -qhalt=w)
+if(OMR_HOST_ARCH STREQUAL "ppc")
+	set(OMR_WARNING_AS_ERROR_FLAG -qhalt=w)
 
-#There is no enhanced warning for XLC right now
-set(OMR_ENHANCED_WARNING_FLAG )
+	# There is no enhanced warning for XLC right now
+	set(OMR_ENHANCED_WARNING_FLAG )
 
-list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
-	-qalias=noansi
-	-qxflag=LTOL:LTOL0
-)
+	list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
+		-qalias=noansi
+		-qxflag=LTOL:LTOL0
+	)
 
-list(APPEND OMR_PLATFORM_CXX_COMPILE_OPTIONS -qlanglvl=extended0x)
+	list(APPEND OMR_PLATFORM_CXX_COMPILE_OPTIONS -qlanglvl=extended0x)
 
-if(OMR_ENV_DATA64)
-	list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
-		-q64
+	if(OMR_ENV_DATA64)
+		list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
+			-q64
+		)
+	else()
+		# -qarch should be there for 32 and 64 C/CXX flags but the C compiler is used for the assembler and it has trouble with some assembly files if it is specified
+		list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
+			-q32
+			-qarch=ppc
+		)
+	endif()
+
+	# Testarossa build variables. Longer term the distinction between TR and the rest
+	# of the OMR code should be heavily reduced. In the mean time, we keep
+	# the distinction
+
+	# TR_COMPILE_OPTIONS are variables appended to CMAKE_{C,CXX}_FLAGS, and so
+	# apply to both C and C++ compilations.
+	list(APPEND TR_COMPILE_OPTIONS
+		-qarch=pwr7
+		-qtls
+		-qnotempinc
+		-qenum=small
+		-qmbcs
+		-qfuncsect
+		-qsuppress=1540-1087:1540-1088:1540-1090:1540-029:1500-029
+		-qdebug=nscrep
 	)
-else()
-	#-qarch should be there for 32 and 64 C/CXX flags but the C compiler is used for the assembler and it has trouble with some assembly files if it is specified
+
+	# Configure the platform dependent library for multithreading
+	set(OMR_PLATFORM_THREAD_LIBRARY -lpthread)
+elseif(OMR_HOST_ARCH STREQUAL "s390")
```

Comment:
This is in the xlc CMake tool configuration. Although I cannot imagine why you would attempt to build OMR on Linux with xlc it technically is an option. I don't see why the same configuration we use for xlc on z/OS is not wanted on xlc for Linux.

Reply:
Note OMR currently only supports building with gcc on Linux on z.

Reply:
just because we aren't doing it/its not supported deosnt mean that somebody wont try it.
Should be easy enough to change from `elseif(OMR_HOST_ARCH STREQUAL "s390")` to `elseif(OMR_HOST_OS STREQUAL "zos")`